extensions/arc/runtime/src/main/java/io/quarkus/arc/properties/IfBuildProperty.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.arc.properties;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * When applied to a bean class or producer method (or field), the bean will only be enabled
+ * if the Quarkus build time property matches the provided value.
```

Comment:
I'd also add a line that mentions that by default if the property is missing you don't enable the bean but it's configurable.

Reply:
Done